code/modules/mob/living/silicon/robot/inventory.dm

```diff
@@ -233,7 +233,7 @@
 
 /mob/living/silicon/robot/unEquip(obj/item/I)
 	if(I == module_active)
-		deselect_module(get_selected_module())
+		uneq_active(I)

```

Comment:
Even when I solved the problem of borgs not being able to regen their stacks, I found that the item was still being referenced by the module slot it was held in; this fixes that.